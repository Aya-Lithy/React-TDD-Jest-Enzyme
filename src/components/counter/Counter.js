import React from "react";

const Counter = () => {
  const [counter, setCounter] = React.useState(0);

  return (
    <div className="counter">
      <h1>This is counter app</h1>
      <div id="counter-value">{counter}</div>
      <button id="increment-btn" onClick={() => setCounter(counter + 1)}>
        increment
      </button>
    </div>
  );
};

export default Counter;
