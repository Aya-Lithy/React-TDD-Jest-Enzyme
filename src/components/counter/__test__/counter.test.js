import { cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Counter from "../Counter";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Counter />);
});

afterEach(cleanup);

test("render the title of the counter", () => {
  //console.log(wrapper.debug());
  expect(wrapper.find("h1").text()).toContain("This is counter app");
});

test("render a button with text of increment", () => {
  expect(wrapper.find("#increment-btn").text()).toBe("increment");
});

test("render the initial value of state in a div", () => {
  expect(wrapper.find("#counter-value").text()).toBe("0");
});

test("render the click event of increment btn & increment counter value", () => {
  wrapper.find("#increment-btn").simulate("click");
  expect(wrapper.find("#counter-value").text()).toBe("1");
});

//snapshot testing
it("matches snapshot", () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
});
