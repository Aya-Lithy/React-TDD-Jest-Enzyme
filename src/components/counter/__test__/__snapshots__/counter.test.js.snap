// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="counter"
>
  <h1>
    This is counter app
  </h1>
  <div
    id="counter-value"
  >
    0
  </div>
  <button
    id="increment-btn"
    onClick={[Function]}
  >
    increment
  </button>
</div>
`;
